src/postamble.js

```diff
@@ -44,8 +43,15 @@ if (memoryInitializer) {
   } else if (Module['memoryInitializerPrefixURL']) {
     memoryInitializer = Module['memoryInitializerPrefixURL'] + memoryInitializer;
   }
-  if (ENVIRONMENT_IS_NODE || ENVIRONMENT_IS_SHELL) {
-    var data = Module['readBinary'](memoryInitializer);
+  var data = Module['memoryInitializerBytes'];
+  if (!data) {
+    if (ENVIRONMENT_IS_NODE || ENVIRONMENT_IS_SHELL) {
+      data = Module['readBinary'](memoryInitializer);
+    } else {
+      data = tryParseAsDataURI(memoryInitializer);
```

Comment:
it looks like we get here in the web case. that means that we parse the data URI instead of passing it to an xhr and letting the browser do so, i think?